custom/icds/rules/custom_actions.py

```diff
@@ -111,7 +114,10 @@ def escalate_tech_issue(case, rule):
 
     num_creates = 0
     num_related_updates = 0
-    tech_issue_delegate = _get_escalated_tech_issue_delegate(case, current_location_id)
+    if toggles.ICDS_AUTO_ESCALATION_QA.enabled(case.domain):
+        tech_issue_delegate = _get_escalated_tech_issue_delegate(case, current_location_id)
+    else:
+        tech_issue_delegate = _get_escalated_tech_issue_delegate(case, escalated_location_id)
```

Comment:
It escalates when the flag is off? Or `current_location_id` is already the escalated one when the flag is on?

Reply:
The former. The bug this is fixing is that the escalation workflow is currently creating a new delegate case every time the original case is escalated, whereas it should reuse the same delegate. This code attempts to find the delegate but hasn't been finding it because it's looking for a delegate with the new/escalated level, when the delegate is still at the current level.